### ISSUE TITLE
add MBTiles support

### ISSUE THREAD

**Issue by incanus** [mapbox/mbxmapkit#3]
This should be straightforward based on [RMMBTilesSource](https://github.com/mapbox/mapbox-ios-sdk/blob/release/MapView/Map/RMMBTilesSource.h). However, linking to SQLite is a problem if we want to keep things simple for installation. Two thoughts right now: 
1. For CocoaPods, make MBTiles suppert a [subspec](http://docs.cocoapods.org/specification.html#subspec). Linking isn't much of a big deal here anyway since it's managed. 
2. For manual install, just make a macro at the top of the file to `#ifdef` out the SQLite code by default. Then, if the dev wants, they can flip the macro bit and link it themselves. This gets around people having to link before use.

**Comment by wsnook** [comment]
It's been a while since this issue was last updated, so for anyone who's still watching, I recently made another MBTiles pull request, #142, which adds a simple and relatively polished MBTiles overlay with overzooming.

I wrote this code offline for use in an app that's not on GitHub. So, the reason I'm creating a new PR (vs. modifying an existing one) is that I already had the code working pretty well and I wanted to make it publicly available. `MBXMBTilesOverlay` is pretty self-contained, so even if #142 doesn't get merged, it should be pretty easy to copy and paste into other projects.
